### ISSUE TITLE
Multi user install flake needs to be a git repo

### ISSUE THREAD

**Issue by rolfst** [NixOS/nix#8091]
**Describe the bug**

on a linux system where I installed nix **without** `--daemon` and then issued command `nix flake init -t '<template_location>'` I can do `nix run` without any issues however when I install nix on a linux system with `--daemon` the same flake init and then do nix run I get an error error: `'<home_dir>' is not a Git repository`. Even when I do `nix run -vvvvv` I get no extra output. i
Only when I add the flake.nix and any resources required by the flake to git the flake will create it's derivation.

If you have a problem with a specific package or NixOS,
you probably want to file an issue at https://github.com/NixOS/nixpkgs/issues. 

**Steps To Reproduce**

1. install Nix with following on a Linux system (other than NixOS of course): 
`
VERSION='2.13.3'
URL="https://releases.nixos.org/nix/nix-${VERSION}/install"
sh <(curl --location "${URL}") \
 --daemon \
 --no-channel-add`
3. initialize a new flake: `nix flake init --template '<template_location>'`
4. Issue command: `nix run`
5. See error: `'<home_dir>' is not a Git repository`

**Expected behavior**

The execution of the flake outputs

**Extra Information**

when installed with:
`
VERSION='2.13.3'
URL="https://releases.nixos.org/nix/nix-${VERSION}/install"
sh <(curl --location "${URL}") \
 --no-daemon \
 --no-channel-add`
`
or 
`
VERSION='2.13.3'
URL="https://releases.nixos.org/nix/nix-${VERSION}/install"
sh <(curl --location "${URL}") \
 --no-channel-add`
`
the flake doesn't need to be a git repo.

**Comment by balsoft** [comment]
Can you try `nix run .` or `nix run path:.` instead?

**Comment by rolfst** [comment]
`nix run path:.` is the only option that works `nix run .` doesn't

**Comment by balsoft** [comment]
Ah, I think I might've seen this one before. Do you by any chance have a `.git` directory in your home dir?

**Comment by rolfst** [comment]
Yes I have. ah so it is a known issue. but the `path:.` makes sure it doesn't break out of the current directory.
I will look what happens when I remove the `.git` folder from my home directory
